2_process/src/process_and_style.R

```diff
@@ -1,19 +1,9 @@
-process_data <- function(nwis_data){
+process_data <- function(nwis_data, site_filename){
   nwis_data_clean <- rename(nwis_data, water_temperature = X_00010_00000) %>% 
     select(-agency_cd, -X_00010_00000_cd, -tz_cd)
-  
-  return(nwis_data_clean)
-}
-
-annotate_data <- function(site_data_clean, site_filename){
   site_info <- read_csv(site_filename)
-  annotated_data <- left_join(site_data_clean, site_info, by = "site_no") %>% 
+  annotated_data <- left_join(nwis_data_clean, site_info, by = "site_no") %>% 
     select(station_name = station_nm, site_no, dateTime, water_temperature, latitude = dec_lat_va, longitude = dec_long_va)
-  
-  return(annotated_data)
-}
-
-
-style_data <- function(site_data_annotated){
-  mutate(site_data_annotated, station_name = as.factor(station_name))
-}
\ No newline at end of file
+  site_data_styled <- mutate(annotated_data, station_name = as.factor(station_name))
+  return(site_data_styled)
+  }
```

Comment:
Hi Amelia - this all looks great, and no need for changes here, but just wanted to mention that you could chain all of this together, rather than having as many intermediate objects. The whole pipe (`%>%`) thing was new to me coming to R from python, so I figure it may be to you as well. So for example, we could reduce this to fewer lines of codes and intermediate objects as follows:

```r
process_data <- function(nwis_data, site_filename) {
  site_info <- read_csv(site_filename)
  nwis_data_clean <- nwis_data %>%
      rename(water_temperature = X_00010_00000) %>% 
      select(-agency_cd, -X_00010_00000_cd, -tz_cd) %>%
      left_join(site_info, by = "site_no") %>% 
      select(station_name = station_nm, site_no, dateTime, water_temperature, latitude = dec_lat_va, longitude = dec_long_va) %>%
      mutate(station_name = as.factor(station_name))
}
```

Reply:
Ah, interesting! I was not familiar with piping, but I will try to practice using it moving forward if I see a good opportunity! Thanks for the suggestion.

Reply:
Yeah! There a really handy [feature](https://style.tidyverse.org/pipes.html) in the `tidyverse` universe.